types/node/index.d.ts

```diff
@@ -81,10 +81,16 @@ declare var __filename: string;
 declare var __dirname: string;
 
 declare function setTimeout(callback: (...args: any[]) => void, ms: number, ...args: any[]): NodeJS.Timer;
+declare namespace setTimeout {
+    export function __promisify__(ms: number): Promise<void>;
```

Comment:
According to https://nodejs.org/api/timers.html#timers_settimeout_callback_delay_args, you can pass a value as well: `setTimeoutPromise(40, "foobar")`, so you might want to consider:
```ts
declare namespace setTimeout {
  export function __promisify__(ms: number): Promise<void>;
  export function __promisify__<T>(ms: number, value: T): Promise<T>;
}
```